chrome/content/zotero/xpcom/style.js

```diff
@@ -399,6 +431,72 @@ Zotero.Styles = new function() {
 			}
 		});
 	}
+	
+	/**
+	 * Populate menulist with locales
+	 */
+	this.populateLocaleList = function(menulist, prefLocale) {
+		if(!_initialized) this.init();
+		
+		var doc = menulist.ownerDocument;
+		var popup = doc.createElement('menupopup');
+		menulist.appendChild(popup);
+		
+		var desiredLocale = "";
+		var fallbackLocale = Zotero.locale;
+		
+		// Primary dialect conversion (e.g. "en" to "en-US")
+		if (Zotero.Styles.primaryDialects[prefLocale] !== undefined) {
+			prefLocale = Zotero.Styles.primaryDialects[prefLocale];
+		}
+		if (Zotero.Styles.primaryDialects[fallbackLocale] !== undefined) {
+			fallbackLocale = Zotero.Styles.primaryDialects[fallbackLocale];
+		}
+		
+		if (prefLocale) {
+			desiredLocale = prefLocale;
+		} else {
+			desiredLocale = fallbackLocale;
+		}
+		
+		var menuLocales = {};
+		var menuLocalesKeys = [];
+		var styleLocales = Zotero.Styles.locales;
+		
+		for (let locale in styleLocales) {
+			if (styleLocales[locale] !== undefined) {
```

Comment:
Actually, that doesn't create a deep copy. Zotero does have
https://github.com/zotero/zotero/blob/c795cf5d82414912d877b60c4dd372f3e53a05e5/chrome/content/zotero/xpcom/utilities.js#L991
though

On Wed, May 6, 2015 at 8:51 PM, Aurimas Vinckevicius <aurimas.dev@gmail.com>
wrote:

> Sorry about that. It should have been just Object.create(Zotero.Styles.locales),
> but whatever you have is fine too.
>
> On Wed, May 6, 2015 at 8:21 PM, Rintze M. Zelle <notifications@github.com>
> wrote:
>
>> In chrome/content/zotero/xpcom/style.js
>> <https://github.com/zotero/zotero/pull/623#discussion_r29818144>:
>>
>> > +		if (Zotero.Styles.primaryDialects[fallbackLocale] !== undefined) {
>> > +			fallbackLocale = Zotero.Styles.primaryDialects[fallbackLocale];
>> > +		}
>> > +		
>> > +		if (prefLocale) {
>> > +			desiredLocale = prefLocale;
>> > +		} else {
>> > +			desiredLocale = fallbackLocale;
>> > +		}
>> > +		
>> > +		var menuLocales = {};
>> > +		var menuLocalesKeys = [];
>> > +		var styleLocales = Zotero.Styles.locales;
>> > +		
>> > +		for (let locale in styleLocales) {
>> > +			if (styleLocales[locale] !== undefined) {
>>
>> Taken care of in rmzelle@6dacf7b
>> <https://github.com/rmzelle/zotero/commit/6dacf7b5e8950c0fa09ce11246f9c1f66f9fa5e0>
>> . For some reason I couldn't get Object.create({}, Zotero.Styles.locales)
>> to work.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/zotero/zotero/pull/623/files#r29818144>.
>>
>
>